### PR TITLE
Make selectString consistent across JsonReaders.

### PR DESCRIPTION
**Pull request by NightlyNexus**
Make JsonValueReader.selectString return -1 for non-strings instead of throwing.

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/JsonValueReader.java [hunk 1]
```diff
@@ -168,9 +168,20 @@
   }
 
   @Override public int selectString(Options options) throws IOException {
-    String peeked = require(String.class, Token.STRING);
+    Object peeked = (stackSize != 0 ? stack[stackSize - 1] : null);
+    String peekedString;
+
+    if ((peeked instanceof String)) {
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/JsonValueReader.java]: nit: extra parens?

Comment (swankjesse) [reply moshi/src/main/java/com/squareup/moshi/JsonValueReader.java]: or, if you invert this if, `if (!(peeked instance of String)) { ... }` you can avoid separate declaration & assignment for peekedString

**Comment by swankjesse** [comment]
Good find!

File: moshi/src/main/java/com/squareup/moshi/JsonValueReader.java (hunk 1 above)
Comment (NightlyNexus) [reply moshi/src/main/java/com/squareup/moshi/JsonValueReader.java]: Yep, thanks!

**Pull request merged**